File: 200-299/206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
	#recursive approach
    def reverseList(self, head: 'ListNode') -> 'ListNode':

        if not head or not head.next:
            return head

        cur = self.reverseList(head.next)
        head.next.next = head;
        head.next = None;
        
        return cur
        

File: 200-299/206-reverse-linked-list.java
/**
 * Definition for singly-linked list
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {val = x; }
 * }
 */

class Solution {
    //recursive approach
    public ListNode reverseList(ListNode head) {
        
        ListNode cur = new ListNode(0);
        if (head == null || head.next == null) {
            return head;
        }
        
        cur = reverseList(head.next);
        head.next.next = head;
        head.next = null;
        
        return cur;
    }
}